### ISSUE TITLE
bind 9.10 exports differently?

### ISSUE THREAD

**Issue by miekg** [digitalocean/bind_exporter#7]
Hi, I'm running this with bind 9.10: 1:9.10.3.dfsg.P2-4 under ubuntu, and there is barfs with:

```
ERRO[0028] Failed to unmarshal XML response: expected element type <isc> but have <statistics>  source=bind_exporter.go:219
```

**Comment by miekg** [comment]
hmm, beginning to assume they've changed the entire xml output.

**Comment by fanf2** [comment]
Yes, 9.10 has a new schema - see https://source.isc.org/cgi-bin/gitweb.cgi?p=bind9.git;a=blob;f=CHANGES;hb=refs/heads/v9_10#l2667

**Comment by miekg** [comment]
@fanf2 where is the schema/definition for this new xml? Do you have any links?

**Comment by grobie** [comment]
I'd love to add tests to the bind exporter. There is already one example xml file in examples, let's move it to a `fixtures` folder and add one for bind 9.10 as well.

**Comment by fanf2** [comment]
@miekg I haven't looked at this in much detail I'm afraid. As far as I can tell from some brief grepping there isn't a formal schema. The xml generation code is in [statschannel.c](https://source.isc.org/cgi-bin/gitweb.cgi?p=bind9.git;a=blob;f=bin/named/statschannel.c;hb=HEAD) and there is a stylesheet in [bind9.xsl](https://source.isc.org/cgi-bin/gitweb.cgi?p=bind9.git;a=blob;f=bin/named/bind9.xsl;hb=HEAD) but neither of them are ideal.

I have a stalled project to get collectd working; a stumbling block is my servers are running Ubuntu 14.04 LTS which has an old version of collectd without BIND 9.10 xml support :-( which is beside the point but you can look at https://github.com/collectd/collectd/blob/master/src/bind.c for an example of a parser that handles v2 and v3 BIND stats. Also rather less than ideal.

**Comment by miekg** [comment]
@fanf2 
> As far as I can tell from some brief grepping there isn't a formal schema.

Mind blown... I'll take a look at your collectd later and hopefully will come up with a pr or something.

**Comment by fanf2** [comment]
@miekg The collectd reader isn't mine - I have just looked at it a bit :-) Good luck getting something working...

**Comment by grobie** [comment]
This has been fixed with #11 and can be closed @mattkanwisher @cagedmantis.
